### DISCUSSION TITLE
Problems with Epub

### DISCUSSION THREAD

**Discussion by LisaSh16** [notesjor/corpusexplorer2.0#13] [Q&A] [answered]
Hello! I am quite new to the corpusexplorer and have problems with creating the corpus for my project. I have a sample of ebooks I'd like to analyze but somehow the epub format does not work. When I upload the texts in epub format the text space stays just blank and no text appears. Could someone please help me? Thanks!

**Comment by notesjor** [comment, marked as answer]
Unfortunately, epub is a relatively complex format. Please try the following:
- Is the output really empty? - Very often the actual text is preceded by a lot of blank lines when reading epub. Therefore, please scroll first and see if the text does not appear. Blank lines may be annoying in the preview, but they do not matter for further processing and are removed automatically.
- There are several add-ons that support epub in other 'flavors'. When CorpusExplorer starts (home page) you will find a list of add-ons at the bottom. You can click on 'Installieren' (install) there. Confirm the request with 'Ja' (yes). After that restart CorpusExplorer. I recommend the following plugins (after '>' you will find the file format for 'annotate documents'/:
    - PANDOC > Pandoc [epub]
    - Tika > Auf gut Glück mit Apache Tika
- Please keep in mind that epubs may be protected with tricks against unauthorized copying/reading. You may not get a satisfying result even with the mentioned add-ons. If you can, you should (generally always) use transparent formats (e.g. XML, TXT).
